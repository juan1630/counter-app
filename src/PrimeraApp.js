// FC
// import React, { Fragment} from 'react';
// <></> Fragments cortos

import React from 'react';
import PropTypes from 'prop-types';

// no se pueden imprimir objectos ni bolleanos
// const saludo = [1,2,3,4,5,6,7];
// const saludo = {
//   nombre:"Juan",
//   edad:25
// }

// usamos el destructurins con los props
const PrimeraApp = ( {
                        saludo,
                        subtitulo
                       } ) => {
  // los props

  // const saludo = "Hola mundo";
  // console.log( props.saludo );
  // esta sintaxis solo acepta valores primitivos
  // <h1> { JSON.stringify( saludo ) } </h1>
   // { JSON.stringify(saludo, null, 3)}
// no imprime booleans
    return (
      <>
      <h1> {saludo} !!!</h1>
      <p>{ subtitulo}</p>
      </>
  );
}


PrimeraApp.propTypes = {
    saludo:PropTypes.string.isRequired
}

//default props
PrimeraApp.defaultProps = {
  subtitulo: 'Soy un subtitulo'
}
export default PrimeraApp;
