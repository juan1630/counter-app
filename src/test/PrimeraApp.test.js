import React from 'react';

import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import PrimeraApp from '../PrimeraApp';



describe('preubas en el componente primer app', () => {

    // test('Debe mostrar hola soy goku', () => {
    //     const saludo ="Hola soy Gokú";
    //     // cuando son porps requeridas se deb de enviar dentro del componente
    //     const { getByText } = render( <PrimeraApp  saludo={saludo} /> );
    //     // wrapper.getByText();
    //     expect( getByText( saludo ) ).toBeInTheDocument();
    // });


    test('debe de mostrar el componente <PrimeraApp/>', ()=> {

          const saludo = "Hola soy Gokú";
          const wrapper  = shallow( <PrimeraApp saludo={saludo}/> );
          expect(  wrapper ).toMatchSnapshot();
          //
    });

// fin del describe
});
