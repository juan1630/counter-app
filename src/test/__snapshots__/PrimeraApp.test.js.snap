// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preubas en el componente primer app debe de mostrar el componente <PrimeraApp/> 1`] = `undefined`;
  
